google/resource_compute_disk.go

```diff
@@ -13,7 +13,7 @@ import (
 )
 
 const (
-	computeDiskUserRegexString = "^(?:https://www.googleapis.com/compute/v1/projects/)?([-_a-zA-Z0-9]*)/zones/([-_a-zA-Z0-9]*)/instances/([-_a-zA-Z0-9]*)$"
+	computeDiskUserRegexString = "^(?:https://www.googleapis.com/compute/v1/projects/)?([-_a-zA-Z0-9:.]*)/zones/([-_a-zA-Z0-9]*)/instances/([-_a-zA-Z0-9]*)$"
```

Comment:
This is just a little bit too loose: you probably want this regex right here:
https://github.com/terraform-providers/terraform-provider-google/blob/411017c35f96ff73b951e49607093b442cad360a/google/validation.go#L15:2